stable/ambassador/templates/ambassador-pro-service.yaml

```diff
@@ -19,18 +19,39 @@ metadata:
       proto: grpc
       auth_service: 127.0.0.1:{{ .Values.pro.ports.auth }}
       allow_request_body: false # setting this to 'true' allows Plugin and External filters to access the body, but has performance overhead
+      {{- if .Values.env }}
+      {{- range $key,$value := .Values.env }}
+      {{- if eq $key "AMBASSADOR_ID" }}
+      ambassador_id: {{ $value | quote }}
+      {{- end }}
+      {{- end }}
+      {{- end }}
       ---
       # This mapping needs to exist, but is never actually followed.
       apiVersion: ambassador/v1
       kind: Mapping
       name: callback_mapping
       prefix: /callback
       service: NoTaReAlSeRvIcE
+      {{- if .Values.env }}
+      {{- range $key,$value := .Values.env }}
+      {{- if eq $key "AMBASSADOR_ID" }}
+      ambassador_id: {{ $value | quote }}
+      {{- end }}
+      {{- end }}
+      {{- end }}
```

Comment:
I think these could be simplified to 

```
{{- if .Values.env.AMBASSADOR_ID }}
ambassador_id: {{ .Values.env.AMBASSADOR_ID | quote }}
{{- end }}
```

Reply:
alternatively `hasKey .Values.env "AMBASSADOR_ID"` in the if :)

Reply:
Helm is smart enough to iterate through the array and look for `AMBASSADOR_ID` key? Very cool, I did not know that 🙂

Reply:
go template magic, you can also use all functions in sprig with helm templates http://masterminds.github.io/sprig/

Reply:
For example 

```
pluck "AMBASSADOR_ID" .Values.env .Values.pro.env | first | quote
```

should allow having ambassador id specified in either one of the env maps, even though this usecase is probably a bad example as it would be weird to just set it in pro.env 😅